### ISSUE TITLE
Issue with nested classes in workflow data object when using mongoDb

### ISSUE THREAD

**Issue by javierdc** [danielgerlag/workflow-core#481]
If I use a class that has a field with the name Id, I get a deserialization error, to repro just create a workflow data with a nested class that has an Id:

public class WorkflowData
    {
        public WorkflowData() 
        {
            Test = new Test();
        }
        public Test Test { get; set; }
    }

public class Test
    {
        public Test()
        {
            Id = Guid.NewGuid().ToString();
        }
        public string Id { get; set; }
        public string Value1 { get; set; }
    }

Note: to repro you must initialize the nested class or create a step that sets some data into the nested class, what happens is that when the workflow tries to deserialize the workflow data for subsequent steps, a deserialization error will happen.

This is because the BSON deserializer seems to have hardcoded Id and tried to treat it as BsonId, how I work around the issue is by commenting the line 41 in MongoPersistenceProvider.cs, although not sure if this workaround will have side effects.

**Comment by danielgerlag** [comment]
Have you tried registering your class with `BsonClassMap.RegisterClassMap`?

**Comment by javierdc** [comment]
Yes, I tried but I get the same error, so far the only workaround is to remove the line mentioned above.

**Comment by danielgerlag** [comment]
How did you do it? Did you specify a specific serializer for your Id field in your class?

**Comment by javierdc** [comment]
No specific serializer, I already had my classes registered because I also use MongoDb to persist all my data. No issues there, it is just with the custom serializer used by Workflow which seems to have an issue with fields named "Id". The Id field is a string.

**Comment by danielgerlag** [comment]
There's nothing in the custom serializer that would treat `Id` fields differently... perhaps if you are nesting classes that you are already persisting as top level document in Mongo, it's using the same mapping from there?

**Comment by javierdc** [comment]
I also tried creating a new class that is not persisted nor mapped and I get the same result, not sure why the custom serializer does not handle "Id" unless it is explicitly mapped. Another finding is that if I comment out the custom serializer of the workflow, then I have no issues with the Id field. although I don't know what is the side effect for the workflow.

**Comment by wfcore-manager** [comment]
Closing as not planned — this has been open for several years without recent progress. Please open a new issue if this is still relevant on a current Workflow Core release.
